api/dataset.go

```diff
@@ -50,63 +51,117 @@ func (api *API) getDefaultDatasetJSON(ctx context.Context, w http.ResponseWriter
 		"version": params.version,
 	}
 
-	geoDimensions, err := api.getGeographyTypes(ctx, params.basedOn)
+	response, err := api.getDatasetJSON(ctx, r, params)
+
 	if err != nil {
 		api.respond.Error(
 			ctx,
 			w,
 			statusCode(err),
 			Error{
-				err:     errors.Wrap(err, "failed to get geography types"),
+				err:     err,
 				logData: logData,
 			},
 		)
-		return
+	} else {
+		api.respond.JSON(ctx, w, http.StatusOK, response)
 	}
+}
 
-	dimensions, _ := api.sortGeography(geoDimensions, params.unsortedDimensions)
-
+func (api *API) getDatasetJSON(ctx context.Context, r *http.Request, params *datasetParams) (*GetDatasetJSONResponse, error) {
 	datasetRequest := cantabular.StaticDatasetQueryRequest{
 		Dataset:   params.basedOn,
-		Variables: dimensions,
+		Variables: params.sortedDimensions,
+	}
+
+	if r.URL.Query().Get("geography") != "" {
+		filters, err := api.getGeographyFilters(ctx, r, params)
+
+		if err != nil {
+			return nil, errors.Wrap(err, "getGeographyFilters failed")
+		}
+
+		datasetRequest.Filters = filters
 	}
 
 	queryResult, err := api.ctblr.StaticDatasetQuery(ctx, datasetRequest)
 	if err != nil {
-		api.respond.Error(
-			ctx,
-			w,
-			statusCode(err),
-			Error{
-				err:     errors.Wrap(err, "failed to run query"),
-				logData: logData,
-			},
-		)
-		return
+		return nil, errors.Wrap(err, "failed to run query")
 	}
 
 	response, err := api.toGetDatasetJsonResponse(params, queryResult)
 	if err != nil {
-		api.respond.Error(
-			ctx,
-			w,
-			statusCode(err),
-			Error{
-				err:     errors.Wrap(err, "failed to generate response"),
-				logData: logData,
-			},
-		)
-		return
+		return nil, errors.Wrap(err, "failed to generate response")
 	}
 
-	api.respond.JSON(ctx, w, http.StatusOK, response)
+	return response, nil
 }
 
-func (api *API) getGeographyDatsetJSON(ctx context.Context, r *http.Request) (*getDatasetJSONResponse, error) {
-	return &getDatasetJSONResponse{}, nil
+func (api *API) getGeographyFilters(_ context.Context, r *http.Request, params *datasetParams) ([]cantabular.Filter, error) {
+	geographyQuery := strings.Split(r.URL.Query().Get("geography"), ",")
+
+	if len(geographyQuery) < 2 {
+		return nil, errors.New("unable to locate geography")
+	}
+
+	geography := geographyQuery[0]
+	geographyOptions := geographyQuery[1:]
+
+	foundGeography := false
+	for _, datasetGeography := range params.geoDimensions {
```

Comment:
It's preferable to use shorter variable names where possible, especially in places where they are short lived/have limited scope as in a for loop. Something like `for _, d := range..` would be more typical